kernel/src/client/sync/parquet.rs

```diff
@@ -1,12 +1,51 @@
+use std::fs::File;
+
+use arrow_schema::Schema as ArrowSchema;
+use parquet::arrow::arrow_reader::{ArrowReaderMetadata, ParquetRecordBatchReaderBuilder};
 use tracing::debug;
+use url::Url;
 
 use crate::{
-    schema::SchemaRef, DeltaResult, Expression, FileDataReadResultIterator, FileMeta,
-    ParquetHandler,
+    client::{
+        arrow_data::ArrowEngineData,
+        arrow_utils::{generate_mask, get_requested_indices, reorder_record_batch},
+    },
+    schema::SchemaRef,
+    DeltaResult, Error, Expression, FileDataReadResultIterator, FileMeta, ParquetHandler,
 };
 
 pub(crate) struct SyncParquetHandler;
 
+fn try_create_from_parquet(schema: SchemaRef, location: Url) -> DeltaResult<ArrowEngineData> {
+    let file = File::open(
+        location
+            .to_file_path()
+            .map_err(|_| Error::generic("can only read local files"))?,
+    )?;
+    let metadata = ArrowReaderMetadata::load(&file, Default::default())?;
+    let parquet_schema = metadata.schema();
+    let requested_schema: ArrowSchema = (&*schema).try_into()?;
```

Comment:
We have `impl TryFrom<&StructType> for ArrowSchema` in `arrow_conversions.rs`

`schema` is an `Arc<StructType>`. `Arc` implements `Deref` so:
- `*schema` is a `StructType`
- `&*schema` is a `&StructType` which supports `try_into`, so we can do that

This could also be written as `schema.deref().try_into()?` if you think that's cleaner.

Reply:
I only asked because https://arrow.apache.org/rust/parquet/arrow/fn.parquet_to_arrow_schema.html exists, I was curious if you were doing a bit of extra work here. But this looks like it works on a SchemaDescriptor and not a StructType

Reply:
This is the `StructType` within delta kernel, i.e. it's an `Arc<deltakernel::schema::StructType>`

Reply:
Yeah, I realized that after you explained, nevermind then